pkg/plugins/repo/client.go

```diff
@@ -53,7 +53,7 @@ func (c *Client) download(_ context.Context, pluginZipURL, checksum string, comp
 		if err := tmpFile.Close(); err != nil {
 			c.log.Warn("Failed to close file", "err", err)
 		}
-		return nil, fmt.Errorf("%v: %w", "failed to download plugin archive", err)
+		return nil, fmt.Errorf("%v: failed to download plugin archive (%s)", err, pluginZipURL)
```

Comment:
Nit: I think this should be:

```suggestion
		return nil, fmt.Errorf("%w: failed to download plugin archive (%s)", err, pluginZipURL)
```

to make sure the original error is wrapped in the new one, like with the previous code